numpy/lib/tests/test_nanfunctions.py

```diff
@@ -388,6 +410,101 @@ def test_empty(self):
             assert_equal(res, tgt)
 
 
+class TestNanFunctions_CumSumProd(TestCase, SharedNanFunctionsTestsMixin):
+
+    nanfuncs = [np.nancumsum, np.nancumprod]
+    stdfuncs = [np.cumsum, np.cumprod]
+
+    def test_allnans(self):
+        # Check for FutureWarning
+        for f, tgt_value in zip(self.nanfuncs, [0, 1]):
+            with warnings.catch_warnings(record=True) as w:
```

Comment:
*your* context manager? but *I* wrote `assert_no_warnings` -- which I think is the right one here :)